### ISSUE TITLE
While training , nn.CrossEntropyLoss() always  be 0 and Validation cross entropy value always be 1.2740095853805542(same value)

### ISSUE THREAD

**Issue by sugerpopo** [milesial/Pytorch-UNet#133]
I try to do segmentation on my own dataset .It a multi class segmentation . Input image are .png picture 3 x 512 x 512 and the mask of each picture is 1 x 512 x 512.Because this is 8 classes segmentation , the pixel value in mask is 0 ,1 2 3 ...8 (0 is background).
I have already changed  :  net = UNet(n_channels=3, n_classes=8)

Because I met some error , I altered some code :
train.py (1)  : 
I add this line
true_masks=true_masks.squeeze(1)
before
loss = criterion(masks_pred, true_masks)

eval.py (2) : 
I changed 
tot += F.cross_entropy(pred.unsqueeze(dim=0), true_mask.unsqueeze(dim=0)).item()
to 
tot += F.cross_entropy(pred.unsqueeze(dim=0), true_mask).item()

dataset.py (3) : 
I altered last line 
return {'image': torch.from_numpy(img), 'mask': torch.from_numpy(mask)}
to 
return {'image': torch.from_numpy(img).type(torch.FloatTensor), 'mask': torch.from_numpy(mask).type(torch.FloatTensor)}

Finally, I start to run the code in 5 epochs ,but the result show bellow is weird:
I find out the loss item always be 0 (sometimes be very small value ,like ' loss (batch)=5.82e-11 ' ).
Besides, INFO: Validation cross entropy: 1.2740095853805542,never changed.

Could anyone tell me how to fix it?
Thanks !

**Comment by drewnia** [comment]
Unfortunately, I have the same error... As well, after training, the prediction procedure is wrong, because output mask has the shape 3xWxH

**Comment by tim-vdl** [comment]
Hi @sugerpopo, have you tried to run training with n_classes=9 instead of 8?

**Comment by sugerpopo** [comment]
Hi  @tim-vdl  ,I have corrected  n_classes to 9.And I noticed that not only image but also mask have divided by 255. It worked! But the result is not that good.I found out my training loss didn't reduce obviously and the Validation cross entropy also not increase continually.
Could you give me some suggestion?
Thanks

**Comment by 15793723081** [comment]
@sugerpopo Hello, I have encountered The same problem. I used The Carvana Data, and The train_loss was reduced to about 0.2, not decreasing，Val_dice goes up to 0.66, not up.Have you solved your problem?this is my qq number:1335578176,If it is convenient to add friends, thank you!

**Comment by milesial** [comment]
This is outdated after the recent code refactor. Closing.
